software/kallistobustools/count/main.nf

```diff
@@ -0,0 +1,59 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process KALLISTOBUSTOOLS_COUNT {
+    tag "$meta.id"
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::kb-python=0.25.1" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/kb-python:0.25.1--py_0"
+    } else {
+        container "quay.io/biocontainers/kb-python:0.25.1--py_0"
+    }
+
+    input:
+    tuple   val(meta),  path(reads)
+    path    index
+    path    t2g
+    path    t1c
+    path    t2c
+    val     use_t1c
+    val     use_t2c
+
+    output:
+    tuple val(meta), path("*_kallistobustools_count") , optional:false , emit: kallistobustools_count
+    path "*.version.txt"                              , optional:false , emit: version
+    path "*.h5ad"                                     , optional:true  , emit: h5ad
+    path "*.matrix.mtx"                               , optional:true  , emit: cellranger_matrix
+    path "*.genes.tsv"                                , optional:true  , emit: cellranger_barcodes
+    path "*.barcodes.tsv"                             , optional:true  , emit: cellranger_tsv
+    path "*.barcodes.loom"                            , optional:true  , emit: loom
```

Comment:
shouldn't these also be tuples having the meta map?

Reply:
Thanks @KevinMenden for the review! Since I am fairly new to DSL2 and nextflow, I actually don't know... What is the goal of returning tuples with the meta map for all outputs?

Reply:
Basically the `meta` map contains some information about the sample, like whether it is single-stranded or the ID of the sample, so it should be passed down to anything that could possibly use that information